lib/nrf_modem_lib/nrf91_sockets.c

```diff
@@ -482,7 +466,17 @@ static int nrf91_socket_offload_getsockopt(void *obj, int level, int optname,
 	int nrf_optname;
 	struct nrf_timeval nrf_timeo = {0, 0};
 	void *nrf_optval = optval;
-	nrf_socklen_t nrf_optlen = (nrf_socklen_t)*optlen;
+	nrf_socklen_t nrf_optlen = 0;
+
+	/* Dereference only if not NULL,
+	 * also prevent segfault later on
+	 */
+	if (optlen != NULL) {
+		nrf_optlen = (nrf_socklen_t)*optlen;
+	} else {
+		errno = EINVAL;
+		return -1;
+	}
```

Comment:
I would assign `nrf_optlen = optlen` unconditionally, and pass just it to `nrf_getsockopt` so that we can receive the correct error from libmodem in case it is `NULL`. If `nrf_getsockopt` is successful, then `optlen` (and `optval`) must be non-NULL, and you can skip other checks.